scripts/build-renode.sh

```diff
@@ -0,0 +1,265 @@
+#!/bin/bash
+
+if [ "`whoami`" = "root" ]
+then
+	echo "Running the script as root is not permitted"
+	exit 1
+fi
+
+CALLED=$_
+[[ "${BASH_SOURCE[0]}" != "${0}" ]] && SOURCED=1 || SOURCED=0
+
+SCRIPT_SRC=$(realpath ${BASH_SOURCE[0]})
+SCRIPT_DIR=$(dirname $SCRIPT_SRC)
+TOP_DIR=$(realpath $SCRIPT_DIR/..)
+
+if [ $SOURCED = 1 ]; then
+	echo "You must run this script, rather then try to source it."
+	echo "$SCRIPT_SRC"
+	exit 1
+fi
+
+if [ -z "$HDMI2USB_ENV" ]; then
+	echo "You appear to not be inside the HDMI2USB environment."
+	echo "Please enter environment with:"
+	echo "  source scripts/enter-env.sh"
+	exit 1
+fi
+
+# Imports TARGET, PLATFORM, CPU and TARGET_BUILD_DIR from Makefile
+eval $(make env)
+make info
+
+source $SCRIPT_DIR/settings.sh
+
+set -x
+set -e
+
+RENODE_BIN="renode"
+command -v $RENODE_BIN 2>&1 1>/dev/null
+if [ $? -ne 0 ]; then
+	RENODE_VERSION=1.7
+	RENODE_PACKAGE=renode_${RENODE_VERSION}_amd64.deb
+	RENODE_URL=https://github.com/renode/renode/releases/download/v$RENODE_VERSION/$RENODE_PACKAGE
+	RENODE_LOCATION="$BUILD_DIR/renode"
+	RENODE_BIN=$RENODE_LOCATION/opt/renode/bin/Renode.exe
+
+	if [ ! -x $RENODE_BIN ]; then
+		mkdir -p $RENODE_LOCATION
+		(
+			cd $RENODE_LOCATION
+			wget $RENODE_URL
+			ar x $RENODE_PACKAGE
+			tar -xf data.tar.gz
+		)
+		
+		chmod u+x $RENODE_BIN
+		echo "Renode downloaded and installed locally: $RENODE_BIN" 
+	fi
+fi
+
+script_file=$(mktemp /tmp/litex_buildenv.resc.XXXXXX)
```

Comment:
Can we move the code generating the configuration to a Python script? It would be much nicer than this then?

Reply:
I also moved generation of resc commands to the same python script.